### PR TITLE
Issue #1298 - Django admin - searchable drop down for User Domain Role

### LINKED ISSUE #1298: Django Admin: Update fields for adding a new "user domain role"
### Issue description

Currently, when adding a "user domain role," the User and Domain fields appear as a standard dropdown, which makes it difficult to find and select the correct entries. This could lead to an unintended selection.

Image

### Acceptance criteria

- [ ] Update the User and Domain dropdowns so that analysts can search for and select an entry 

### Additional context

_No response_

### Links to other issues

Related to #1292 #1291 (Do these at the same time)

### PR DESCRIPTION
**Pull request by dave-kennedy-ecs**
## Ticket

Resolves #1298

## Changes

- Searchable drop down added for user and domain in UserDomainRole Edit and Add in django admin.

## Context for reviewers

This is a change to the interaction when editing (or adding) a UserDomainRole in django admin.  The user and domain drop downs have now been converted to searchable drop downs.

To test, navigate to django admin, and click on UserDomainRole.  Either click Add or click a UserDomainRole from the list to edit.  When the form loads, test the interaction for user and domain.  A screenshot is inserted below

#### Satisfied acceptance criteria and met development standards

- [ ] Met the acceptance criteria, or will meet them in a subsequent PR
- [ ] Created/modified automated tests
- [ ] Added at least 2 developers as PR reviewers (only 1 will need to approve)
- [ ] Messaged on Slack or in standup to notify the team that a PR is ready for review
- [ ] Changes to “how we do things” are documented in READMEs and or onboarding guide
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the assoicated migrations file has been commited.

#### Ensured code standards are met (Original Developer)

- [ ] All new functions and methods are commented using plain language
- [ ] Did dependency updates in Pipfile also get changed in requirements.txt?
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values

#### Validated user-facing changes (if applicable)

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [ ] Add at least 1 designer as PR reviewer

#### Reviewed, tested, and left feedback about the changes

- [x] Pulled this branch locally and tested it
- [x] Reviewed this code and left comments
- [x] Checked that all code is adequately covered by tests
- [ ] Made it clear which comments need to be addressed before this work is merged
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the assoicated migrations file has been commited.

#### Ensured code standards are met (Code reviewer)

- [x] All new functions and methods are commented using plain language
- [x] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values
- [ ] (Rarely needed) Did dependency updates in Pipfile also get changed in requirements.txt?

#### Validated user-facing changes as a developer

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [x] Checked keyboard navigability
- [x] Meets all designs and user flows provided by design/product
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [x] Tested with multiple browsers, the suggestion is to use ones that the developer didn't (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [x] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

**Note:** Multiple code reviewers can share the checklists above, a second reviewers should not make a duplicate checklist

#### Verified that the changes match the design intention

- [ ] Checked that the design translated visually
- [ ] Checked behavior
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links
- [ ] Tried to break the intended flow

#### Validated user-facing changes as a designer

- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

## Screenshots

image

### CODE REVIEW

File: src/Pipfile [hunk 1]
```diff
@@ -44,4 +44,4 @@ django-webtest = "*"
 types-cachetools = "*"
 boto3-mocking = "*"
 boto3-stubs = "*"
-django-model2puml = "*"
+django-model2puml = "*"
```
Comment (zandercymatics) [inline src/Pipfile:47]: (Before merge) Remove accidental code change

File: src/Pipfile.lock [hunk 2]
```diff
@@ -1685,4 +1685,4 @@
             "version": "==3.0.0"
         }
     }
-}
+}
```
Comment (zandercymatics) [inline src/Pipfile.lock:1688]:
```suggestion
}
```

Comment (abroddrick) [reply src/Pipfile.lock:1688]: +1 to removing the accidental code change

File: src/registrar/admin.py [hunk 3]
```diff
@@ -968,6 +948,28 @@ def has_change_permission(self, request, obj=None):
         return super().has_change_permission(request, obj)
 
 
+class UserDomainRoleAdmin(ListHeaderAdmin):
+    """Custom domain role admin class."""
```
Comment (abroddrick) [inline src/registrar/admin.py:952]:
```suggestion
    """Custom user domain role admin class."""
```

**Review by abroddrick — APPROVED** [review]
Pushed to my branch temporarily and tested, looks amazing!

**Review by Katherine-Osos — APPROVED** [review]

**Pull request merged**